README.md

```diff
@@ -1,8 +1,408 @@
+var singer = new Singer { .. };
+session.Save(singer);
+// This will manually reload all properties from the database, including the generated FullName column.
+session.Refresh(singer);
+```
+
+## Mutations
+Cloud Spanner can use Mutations instead of DML to modify data. Mutations can be faster than DML, but Mutations do not support
+read-your-writes. It is recommended to read [https://cloud.google.com/spanner/docs/dml-versus-mutations](https://cloud.google.com/spanner/docs/dml-versus-mutations)
+if you are not familiar with the differences between Mutations and DML in Cloud Spanner.
+
+The Cloud Spanner NHibernate driver supports executing transactions that will use Mutations instead of DML. These transactions
+do not support read-your-writes. Any modifications will only be readable after the transaction has committed. Manually flushing
+the session during a transaction will also not send any modifications to Cloud Spanner.
+
+```csharp
+using var session = sessionFactory.OpenSession();
+using var transactionWithMutations = session.BeginTransaction(MutationUsage.Always);
+singer = session.Load<Singer>(singerId);
+singer.LastName = "Yates-Fish";
+
+// Committing the transaction will send the Mutation to Cloud Spanner and the updated value should be visible.
+transactionWithMutations.Commit();
+singersWithLastNameYatesFish = session.Query<Singer>().Where(s => s.LastName.Equals("Yates-Fish")).ToList();
+```
+
+# Limitations
+
+## Generated Values for Primary Keys
+Cloud Spanner does not support sequences, identity columns, or other value generators in the database that will
+generate a unique value that could be used as a primary key value. Instead, the best option is to use a client
+side Guid generator for a primary key if your table does not contain a natural primary key.
+
+## Default Values
+Cloud Spanner does not support default values for columns. The `SpannerSingleTableEntityPersister` is however able
+to simulate this. See the example above for commit timestamps that uses this feature.
+
+# Performance Recommendations
+NHibernate supports a wide range of different configurations and mappings. Some of these can lead to a large number
+of DML statements being executed when only one entity is updated, or can cause queries to be generated that need to
+join multiple tables. The following list contains recommendations for how to use NHibernate as efficiently as
+possible with Cloud Spanner.
+
+| Feature              | Recommendation                                                                                                                                                                                                                                               |
+|----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```

Comment:
Based on the discussion in the design doc, should we also say to use a read-only tx if you know your tx only has reads? I don't know if it's possible to configure this.

Reply:
I've added a note for that as well. It is not something that we can configure or provide a default for, but a user can choose to start a read-only transaction instead of a read/write transaction when they know that the transaction will only read.